cpp/include/cudf/io/orc_metadata.hpp

```diff
@@ -38,18 +51,130 @@ namespace io {
  * The following code snippet demonstrates how to read statistics of a dataset
  * from a file:
  * @code
- *  std::string filepath = "dataset.orc";
- *  auto result = cudf::read_orc_statistics(cudf::source_info(filepath));
+ *  auto result = cudf::read_raw_orc_statistics(cudf::source_info("dataset.orc"));
  * @endcode
  *
  * @param src_info Dataset source
  *
- * @return Decompressed ColumnStatistics blobs stored in a vector of vectors. The first element of
- * result vector, which is itself a vector, contains the name of each column. The second element
- * contains statistics of each column of the whole file. Remaining elements contain statistics of
- * each column for each stripe.
+ * @return ORC file statistics and column names.
+ */
+raw_orc_statistics read_raw_orc_statistics(source_info const &src_info);
+
+/**
+ * @brief Base class for different column statistics that include optional min and max values.
+ *
+ * Includes accessors for the min and max values.
+ */
+template <typename T>
+struct minmax_statistics {
+  std::unique_ptr<T> minimum;
+  std::unique_ptr<T> maximum;
+
+  auto has_minimum() const { return minimum != nullptr; }
+  auto has_maximum() const { return maximum != nullptr; }
+  auto *get_minimum() const { return minimum.get(); }
+  auto *get_maximum() const { return maximum.get(); }
+};
+
+/**
+ * @brief Base class for different column statistics that include optional sum value.
+ *
+ * Includes accessors for the sum value.
+ */
+template <typename T>
+struct sum_statistics {
+  std::unique_ptr<T> sum;
+
+  auto has_sum() const { return sum != nullptr; }
+  auto *get_sum() const { return sum.get(); }
+};
+
+struct integer_statistics : minmax_statistics<int64_t>, sum_statistics<int64_t> {
+};
+
+struct double_statistics : minmax_statistics<double>, sum_statistics<double> {
+};
+
+// According to ORC specs, the sum should be signed, but pyarrow uses unsigned value
+struct string_statistics : minmax_statistics<std::string>, sum_statistics<uint64_t> {
+};
+
+struct bucket_statistics {
+  std::vector<uint64_t> count;
+
+  auto *get_count(size_t index) const { return &count.at(index); }
+};
+
+struct decimal_statistics : minmax_statistics<std::string>, sum_statistics<std::string> {
+};
+
+struct date_statistics : minmax_statistics<int32_t> {
+};
+
+struct binary_statistics : sum_statistics<int64_t> {
+};
+
+struct timestamp_statistics : minmax_statistics<int64_t> {
+  std::unique_ptr<int64_t> minimumUtc;
+  std::unique_ptr<int64_t> maximumUtc;
+
+  auto has_minimumUtc() const { return minimumUtc != nullptr; }
+  auto has_maximumUtc() const { return maximumUtc != nullptr; }
```

Comment:
`snake_camelCase`? :stuck_out_tongue: :snake: :camel: should it be `mimimum_utc`?